src/script/sconsign.py

```diff
@@ -407,51 +424,62 @@ def __call__(self, fname):
                     # Nope, that file doesn't even exist, so report that
                     # fact back.
                     print_e = e
-                sys.stderr.write("sconsign: %s\n" % (print_e))
+                sys.stderr.write("sconsign: %s\n" % print_e)
                 return
         except KeyboardInterrupt:
             raise
         except pickle.UnpicklingError:
-            sys.stderr.write("sconsign: ignoring invalid `%s' file `%s'\n" % (self.dbm_name, fname))
+            sys.stderr.write("sconsign: ignoring invalid `%s' file `%s'\n"
+                             % (self.dbm_name, fname))
             return
         except Exception as e:
-            sys.stderr.write("sconsign: ignoring invalid `%s' file `%s': %s\n" % (self.dbm_name, fname, e))
+            sys.stderr.write("sconsign: ignoring invalid `%s' file `%s': %s\n"
+                             % (self.dbm_name, fname, e))
             return
 
         if Print_Directories:
             for dir in Print_Directories:
                 try:
                     val = db[dir]
                 except KeyError:
-                    sys.stderr.write("sconsign: no dir `%s' in `%s'\n" % (dir, args[0]))
+                    err = "sconsign: no dir `%s' in `%s'\n" % (dir, args[0])
+                    sys.stderr.write(err)
                 else:
                     self.printentries(dir, val)
         else:
             for dir in sorted(db.keys()):
                 self.printentries(dir, db[dir])
 
-    def printentries(self, dir, val):
-        print('=== ' + dir + ':')
+    @staticmethod
+    def printentries(dir, val):
+        try:
+            print('=== ' + dir + ':')
```

Comment:
For clarity, this problem turned up on a nearly empty sconsign file, which on Python2, which did not error, displayed this way:
```
=== .:
help: None 0 0
```
Using Python 3, it threw an exception:
```
    print('=== ' + dir + ':')
TypeError: must be str, not bytes
```